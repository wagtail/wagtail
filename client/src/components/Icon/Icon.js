import PropTypes from 'prop-types';
import React from 'react';

/**
 * Abstracts away the actual icon implementation (font icons, SVG icons, CSS sprite).
 * Provide a `title` as an accessible label intended for screen readers.
 */
const Icon = ({ name, className, title }) => (
  <span>
    <span className={`icon icon-${name} ${className || ''}`} aria-hidden></span>
    {title ? (
      <span className="visuallyhidden">
        {title}
      </span>
    ) : null}
  </span>
);

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  className: PropTypes.string,
  title: PropTypes.string,
};

Icon.defaultProps = {
  className: null,
  title: null,
};

export default Icon;
