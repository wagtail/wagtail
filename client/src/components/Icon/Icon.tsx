import * as React from 'react';

export interface IconProps {
  name: string;
  className?: string;
  title?: string;
}

/**
 * Provide a `title` as an accessible label intended for screen readers.
 */
const Icon: React.FunctionComponent<IconProps> = ({ name, className, title }) => (
  <>
    <svg className={`icon icon-${name} ${className || ''}`} aria-hidden="true">
      <use href={`#icon-${name}`} />
    </svg>
    {title &&
      <span className="visuallyhidden">
        {title}
      </span>
    }
  </>
);

export default Icon;
