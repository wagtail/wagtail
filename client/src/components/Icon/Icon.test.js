import React from 'react';
import { shallow } from 'enzyme';
import Icon from './Icon';

describe('Icon', () => {
  it('exists', () => {
    expect(Icon).toBeDefined();
  });

  it('#name', () => {
    expect(shallow(<Icon name="test" />)).toMatchSnapshot();
  });

  it('#className', () => {
    expect(shallow(<Icon name="test" className="u-test" />)).toMatchSnapshot();
  });

  it('#title', () => {
    expect(shallow(<Icon name="test" title="Test title" />)).toMatchSnapshot();
  });
});
