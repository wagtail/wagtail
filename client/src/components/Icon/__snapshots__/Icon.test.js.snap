// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon #className 1`] = `
<span>
  <span
    aria-hidden={true}
    className="icon icon-test u-test"
  />
</span>
`;

exports[`Icon #name 1`] = `
<span>
  <span
    aria-hidden={true}
    className="icon icon-test "
  />
</span>
`;

exports[`Icon #title 1`] = `
<span>
  <span
    aria-hidden={true}
    className="icon icon-test "
  />
  <span
    className="visuallyhidden"
  >
    Test title
  </span>
</span>
`;
